agent/proto/resourcesmonitor/fetcher.go

```diff
@@ -0,0 +1,61 @@
+package resourcesmonitor
+
+import (
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/clistat"
+)
+
+type ResourcesFetcher interface {
```

Comment:
this can just be called `Fetcher` with methods `FetchMemory` and `FetchVolume` -- the package name is resourcesmonitor, and Go style is to reduce "stutter" as much as possible.